CleverTap/Runtime/Native/UnityNativeWrapper/Models/UnityNativeARPResponseInterceptor.cs

```diff
@@ -0,0 +1,130 @@
+using System;
+using System.Collections.Generic;
+using CleverTapSDK.Native;
+using CleverTapSDK.Utilities;
+
+#if (!UNITY_IOS && !UNITY_ANDROID) || UNITY_EDITOR
+namespace CleverTapSDK.Native
+{
+    internal class UnityNativeARPResponseInterceptor : IUnityNativeResponseInterceptor
+    {
+        private readonly UnityNativeEventValidator _eventValidator;
+        private readonly string _accountId;
+        private readonly string _namespaceARPKey;
+        public UnityNativeARPResponseInterceptor(string accountId,string deviceId, UnityNativeEventValidator eventValidator)
+        {
+            this._accountId = accountId;
+            this._namespaceARPKey = string.Format(UnityNativeConstants.Network.ARP_NAMESPACE_KEY,accountId,deviceId); 
+            _eventValidator = eventValidator;
+        }
+
+        UnityNativeResponse IUnityNativeResponseInterceptor.Intercept(UnityNativeResponse response)
+        {
+            var result = Json.Deserialize(response.Content) as Dictionary<string, object>;
+            try
+            {
+                if (result.ContainsKey("arp"))
+                    if (Json.Deserialize(result["arp"].ToString()) is Dictionary<string, object> { Count: > 0 } arp)
+                    {
+                        //Handle Discarded events in ARP
+                        try
+                        {
+                            ProcessDiscardedEventsList(arp);
+                        }
+                        catch (Exception t)
+                        {
+                            CleverTapLogger.Log("Failed to process ARP discarded events");
+                        }
+
+                        HandleARPUpdate(arp);
+                    }
+            }
+            catch (Exception exception)
```

Comment:
The 'exception' variable is not used. Please log the exception details.

Reply:
Logged the exception